dbt/adapters/athena/impl.py

```diff
@@ -38,11 +38,64 @@ def convert_datetime_type(
         return "timestamp"
 
     @available
-    def s3_uuid_table_location(self):
+    def s3_table_prefix(self) -> str:
+        """
+        Returns the root location for storing tables in S3.
+
+        This is `s3_data_dir`, if set, and `s3_staging_dir/tables/` if not.
+
+        We generate a value here even if `s3_data_dir` is not set,
+        since creating a seed table requires a non-default location.
+        """
         conn = self.connections.get_thread_connection()
-        client = conn.handle
+        creds = conn.credentials
+        if creds.s3_data_dir is not None:
+            return creds.s3_data_dir
+        else:
+            return f"{creds.s3_staging_dir}tables/"
```

Comment:
If `s3_staging_dir ` has a trailing slash or it miss a / we can have issues. We can do simply:
`os.path.join(creds.s3_staging_dir, 'tables/')`